### PR TITLE
Scala Native: run tests

### PR DESCRIPTION
**Pull request by sideeffffect**
continuation of https://github.com/zio/zio/pull/6283

### CODE REVIEW

**Comment by sideeffffect** [comment]
@lolgab so the trick with `scala.scalanative.loop.EventLoop.run()` in `OneShot` kinda "works" -- the tests _are_ running -- but it's impossibly slow...
Any idea what's going wrong or how to fix that?

**Comment by lolgab** [comment]
@sideeffffect Good trick checking if the value is `null` before doing any `EventLoop.run`. I'm thinking of another EventLoop design that can be totally transparent to users, but still, there is nothing out there. For now, your workaround seems fine.

**EDIT:**
There is a solution to this problem in [cats-effect](https://github.com/typelevel/cats-effect/blob/55e963a5aa742d7e6e0c8c086f5b8e8c868280c2/core/native/src/main/scala/cats/effect/unsafe/PollingExecutorScheduler.scala#L40-L48). When I say that there is nothing out there to solve this issue, I meant there is nothing implemented in scala-native-loop yet to solve this issue.

**Comment by sideeffffect** [comment]
@lolgab could you please publish scala-native-crypto also for Scala 2.11? :pray: https://github.com/lolgab/scala-native-crypto/pull/14

**Comment by sideeffffect** [comment]
@lolgab Any idea what to do about this?
```scala
[error] Found 1 missing definitions while linking
[error] Not found Top(scala.Nothing)
[error] 	at /home/runner/work/zio/zio/test-tests/shared/src/test/scala/zio/test/mock/PolyMockSpec.scala:15
```

EDIT:
Reported as https://github.com/scala-native/scala-native/issues/2920 (now fixed)

**Comment by sideeffffect** [comment]
The problem with `Not found Top(scala.Nothing)` has been reported as https://github.com/scala-native/scala-native/pull/2949 (now fixed)

**Comment by jdegoes** [comment]
@sideeffffect I am very eager to have this in to 2.0. Do you think you might get a chance to continue your work here?

Please re-open if ready for another look.

**Comment by sideeffffect** [comment]
@mijicd can you please re-open this PR? :pray:

**Comment by sideeffffect** [comment]
So now we're failing on a single test:
```
[info] - FiberRefSpec
[info]   - Create a new FiberRef with a specified value and check if:
[info]     - the value of all fibers in inherited when running many ZIOs with collectAllPar
```
I would call that a progress :blush:

**Comment by sideeffffect** [comment]
Weird... now ZIO Native tests pass on my computer :thinking:

**Comment by sideeffffect** [comment]
Wow, this is a very important milestone! With only two minuscule changes to the test suite, we're passing all the CI for ZIO 1.
It seems like ZIO 1 on Native isn't as fast with Fibers and can't create as many of them as on JVM/JS. But from functionality point of view, it seems to be 100 % correct :tada: 

Open question is what to do with it now. Lower the number of fibers in tests for all platforms? Or just for Native? Or find out and fix the reason for the slowness on Native?

The current version, ZIO 2, on Native is a different story, not yet explored.

**Comment by swoogles** [comment]
> Wow, this is a very important milestone! With only two minuscule changes to the test suite, we're passing all the CI for ZIO 1.
> It seems like ZIO 1 on Native isn't as fast with Fibers and can't create as many of them as on JVM/JS. But from functionality point of view, it seems to be 100 % correct :tada: 
> 
> Open question is what to do with it now. Lower the number of fibers in tests for all platforms? Or just for Native? Or find out and fix the reason for the slowness on Native?
> 
> The current version, ZIO 2, on Native is a different story, not yet explored.

Great! Anything else need to be done here?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -181,7 +181,7 @@ jobs:
 
   testPlatforms:
     runs-on: ubuntu-20.04
-    timeout-minutes: 60
+    timeout-minutes: 120
```
Comment (adamgfraser) [inline .github/workflows/ci.yml]: Can we revert this now?

Comment (sideeffffect) [reply .github/workflows/ci.yml]: we can :+1:

**Comment by sideeffffect** [comment]
This can now be merged :tada:

**Review by adamgfraser — APPROVED** [review]

**Comment by jdegoes** [comment]
@sideeffffect Thank you for your work on this! I hope you continue your work here in 2.0 as many users have moved to 2.0 already.

**Comment by sideeffffect** [comment]
Already on it :muscle: 
 * https://github.com/zio/zio/pull/7883

**Pull request merged**